_posts/2022-01-07-Basics-of-Machine-Learning.md

```diff
@@ -0,0 +1,58 @@
+They take the inputs, process them and then generate an output. <br>
+The last type of neurons are the yellow one. They are at the end of the network. Their Output is the output of the whole network. <br>
+
+
+All 3 types of the network can be descriped through one model
+
+<img src="https://miro.medium.com/max/428/1*O7YSSqlOdQuNgMrH7J_3dg.png" width="600">
+
+A neuron has multiple inputs. The inputs are the outputs of all neurons in the previous collum.
+These inputs get weights. For example if a input is 0.6 and the weight is 0.1 we take 0.6 * 0.1 and then we have the "real" input. Every input has its own weight. All of the inputs multiplied with the weigth are added together. This sum is now put into a function. The function determines how much output is generated for the given sum. Often the function looks like this:
+
+
+The output of this function (in this case RELU) is the output of the neuron and therefore the input for the next collum of neurons.<br>
+If we have a description of the arrangement of the neurons and a list of all weights we can reproduce the network over and over with the same outputs for the same inputs. And these two informations are the core of a neuronal network. How the Neurons are arrenged and what the weights are. 
+
+## What is Backpropagation
+The structure of a network impacts its perfomance but the best variable to shape it are the weights. If we get the right weights its should be perfect. Obviously getting good weights is hard. I know i said this is without math but we need a little bit. A neuronal network is like a function with millions of parameters(weights) which produces a deterministic result. We can imagen this like a 3D map. 
+
+
+<img src="https://bdtechtalks.com/wp-content/uploads/2020/04/gradient-descent-local-minima.png" width="600">
+
+The lower we get the better our network performs. Finding the lowest point overall (Global Minima) is at the moment impossible. We take a other approach. We start at a random point somewhere. So we just use random weights. Luckily we have a way to find the way which leads us from our point. This is called negative gradiant. If you are intereseted in math you can research this on your own. <br>
+Now lets start learning. We guess some weights and start throwing inputs into the network. We compare the result of the network with the right results. Then we use the negative gradiant and adjust the weights with these metrics. This process is repeated thousand or even millions of times until we find a local minima. If we reach it we cant get out and have to take a look how our network performs. If the accurecy of the network is enough we are finished. When the result are not good enough we have to start again with other starting weights and let the network learn.
+
+## How can we use this knowledge in practice
```

Comment:
Find die Überschrift etwas lang; ist aber auch nur mein Stil. Ich hätte es schlicht `Practical Applications` genannt.

Reply:
Ok, hab jetzt den TExt des absatzes gelesen und denke, dass das keine gute Überschrift wäre.

Vielleicht sowas wie "Frameworks for Getting Started" oder so. kA. Kann man auch so lassen wie jetzt.